### ISSUE TITLE
"Invalid format" errors for the temporary file containing the signed certificate on MacOS

### ISSUE THREAD

**Issue by bryan-srg** [isometry/vault-ssh-plus#57]
I've been trying to introduce this tool to my team members recently - but two of them are running into issues on their MacBooks, where when trying to start a session using vssh, they see the following error:

```
vssh svc-ansible@<redacted>
The authenticity of host '<redacted> (<redacted>)' can't be established.
ED25519 key fingerprint is SHA256:C6yCNYR1imJo0fzejYrt9da/dRkqz/eCUpbJ9unSgAw.
This key is not known by any other names.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '<redacted>' (ED25519) to the list of known hosts.
Authorized uses only. All activity monitored and reported.
Load key "/var/folders/t0/06kdt0rj0g58395yk4mk36140000gn/T/vssh-cert.4063717399": invalid format
svc-ansible@<redacted>'s password:
```

By the time we've gone to look though, the temporary file "/var/folders/t0/06kdt0rj0g58395yk4mk36140000gn/T/vssh-cert.4063717399" has already been removed, so it's not possible to say what's wrong with it.

Underlying OpenSSH version is
```
ssh -V
OpenSSH_9.8p1, LibreSSL 3.3.6 
```
And Vault is:
```
vault -version
Vault v1.18.1 (cgo)
```

vault-ssh-plus environment variables set:
```
export VAULT_ADDR=https://vault.host.name
# Only need this one if your machine doesn't trust internal certs:
export VAULT_SKIP_VERIFY=true
export VAULT_SSH_ROLE=wildcard_modern_role
export VAULT_SSH_PUBLIC_KEY=~/.ssh/id_ed25519_vault.pub
export VAULT_SSH_PATH=ssh-key-signer
```

I should add, I've walked one of the problem users through using just ssh and getting the signed cert from vault manually, and that _does_ work ok.

**Comment by isometry** [comment]
@bryan-srg thanks for the report!

I believe I've seen this issue when the private key is encrypted with a passphrase but is not loaded into `ssh-agent`. Could you confirm whether running `ssh-add ~/.ssh/id_ed25519_vault` first resolves the error? Alternatively, try with a new passphrase-less private key. If this is the issue, and as the private key isn't important to authorisation, you can consider suggesting the affected users regenerate keys without passphrases. Let me know the results!

**Comment by bryan-srg** [comment]
Instead of going down the ssh-agent route, I went with getting them to change to using issued keys instead of signing their pre-existing one. (I don't think this was a feature when I first came across vault-ssh-plus - so my guide for them was still assuming signing was the way to go). This has resolved the problem for them - so I think we can close this.
Thanks for a great bit of software, and thanks again for such a speedy response!
